extensions/rich_text_components/Skillreview/webdriverio.js

```diff
@@ -17,18 +17,25 @@
  * end-to-end testing with WebdriverIO.js
  */
 
+var action = require(process.cwd() + '/core/tests/webdriverio_utils/action.js');
 var objects = require(process.cwd() + '/extensions/objects/webdriverio.js');
 var forms = require(process.cwd() + '/core/tests/webdriverio_utils/forms.js');
 var waitFor = require(
   process.cwd() + '/core/tests/webdriverio_utils/waitFor.js');
 
 var customizeComponent = async function(modal, text, skillDescription) {
-  await objects.UnicodeStringEditor(
-    modal.$('<schema-based-unicode-editor>')
-  ).setValue(text);
   await objects.SkillSelector(
     modal.$('<skill-selector-editor>')
   ).setValue(skillDescription);
+
+  var textEditor = objects.UnicodeStringEditor(
+    modal.$('<schema-based-unicode-editor>'));
+  // Do not change the text if there is already text present as this
+  // RTE component sometimes relies on highlighting text.
+  if (textEditor.getValue() && !text) {
+    return;
+  }
+  await textEditor.setValue(text);
```

Comment:
This behavior doesn't seem right. If the user calls `customizeComponent` with `''` for `text` because they want to clear the text from the editor, the editor won't be changed anymore, right?

I propose that we distinguish between the case where `text=''`, which indicates that the editor should be cleared, and the case where `text` is null, indicating that the editor's text contents should not be changed.

Reply:
Done